.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,29 @@
+name: Pull Request
+
+on:
+  push:
+    branches: master
+  pull_request:
+    types: [opened, reopened, ready_for_review, synchronize]
+    branches: master
+      
+jobs:
+  cancelPreviousRuns:
+    name: Cancel previous runs
+    if: github.event_name == 'push' || !github.event.pull_request.draft
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ secrets.GITHUB_TOKEN }}
+
+  test:
+    name: Build and Run tests
+    if: github.event_name == 'push' || !github.event.pull_request.draft
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run tests
+      run: swift test -v
```

Comment:
Is it necessary to run this on macOS if only Swift tests are executed? Swift is also installed on `ubuntu-latest`.

Reply:
yes, as the tests needs a iOS Simulator, which is only available in macOS.